### PR TITLE
📦 refactor(card): synced with latest shadcn card implementation

### LINKED ISSUE #310: [Review] Card
### Code Quality

- [ ] Follows standalone component pattern
- [ ] Uses signal-based inputs with `input()` function
- [ ] Implements OnPush change detection
- [ ] Uses CVA variants properly
- [ ] Host binding for dynamic classes: `host: { '[class]': 'classes()' }`
- [ ] Proper TypeScript types and interfaces
- [ ] Accessibility attributes (ARIA) implemented

### Developer Experience (DX)

- [ ] API follows ng-zorro patterns and conventions
- [ ] Inputs are minimal and necessary only
- [ ] Outputs are minimal and necessary only
- [ ] Input/output names are intuitive and consistent
- [ ] Component is easy to use and configure
- [ ] No unnecessary complexity in the API

### Testing

- [ ] Unit tests exist and pass
- [ ] Test coverage is adequate
- [ ] Edge cases are covered

### Documentation

- [ ] `overview.md` is complete and clear
- [ ] `api.md` documents all inputs/outputs
- [ ] Demo components showcase main use cases
- [ ] Demo components follow naming conventions
- [ ] Examples match or closely follow shadcn/ui examples
- [ ] Component behavior aligns with shadcn/ui patterns

### File Structure

- [ ] Component file structure follows standards
- [ ] Variants file exists and is properly typed
- [ ] All files are in correct locations
- [ ] Exports are added to `components.ts`

### Styling

- [ ] Uses TailwindCSS v4 properly
- [ ] Design tokens (CSS custom properties) are used
- [ ] Responsive design is implemented
- [ ] Dark mode support (if applicable)

### PR DESCRIPTION
**Pull request by mikij**
## What was done? 📝

Refactored the card component to sync with the latest shadcn card implementation. This update brings the component up to date with current shadcn/ui standards and best practices.

### Core Component Updates
- **Card Component** (`libs/zard/src/lib/shared/components/card/`): Updated the main component implementation, variants, and test suite to align with latest shadcn patterns
- **Tests**: Comprehensive updates to the card component test suite

### Documentation & Demos
- **API Documentation**: Updated component API documentation
- **Demo Files**: Refreshed demo examples and usage patterns
- **Installation Guide**: Updated manual installation instructions for the card component

### Application Integration
- **Component Pages**: Updated demo pages and component showcases
- **Code Examples**: Modified code box component for better card examples
- **Dark Mode Support**: Adjusted dark mode page implementations

### Configuration Updates
- **ESLint**: Updated linting configuration
- **Prettier**: Modified code formatting rules

This refactor ensures the card component stays current with the latest shadcn/ui design system while maintaining backward compatibility and improving overall component quality.

## Screenshots or GIFs 📸

image

## Link to Issue 🔗

Closes #310 

## Type of change 🏗

- [ ] New feature (non-breaking change that adds functionality)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] Refactor (non-breaking change that improves the code or technical debt)
- [ ] Chore (none of the above, such as upgrading libraries)

## Breaking change 🚨

API has been extended. Old behavior should require no changes.

## Checklist 🧐

- [x] Tested on Chrome
- [ ] Tested on Safari
- [ ] Tested on Firefox
- [x] No errors in the console

## Summary by CodeRabbit

* **New Features**
  * Cards gain an optional action button with click events, configurable header/footer borders, and a dedicated footer area.
  * Installation content now uses an accessible tabbed interface; Examples layout updated to match.

* **Accessibility**
  * Card markup includes ARIA relationships and improved keyboard accessibility for interactive elements.

* **Tests**
  * Card tests rewritten to a render-focused approach with broader behavioral and accessibility coverage.

* **Documentation**
  * Card API docs updated to document new inputs, footer usage, and the action event.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: libs/zard/src/lib/shared/components/card/card.component.ts [hunk 1]
```diff
@@ -3,54 +3,95 @@ import {
...
+  readonly zHeaderBorder = input(false);
+  readonly zAction = input('');
   readonly zDescription = input<string | TemplateRef<void>>();
+  readonly zTitle = input<string | TemplateRef<void>>();
 
-  readonly class = input<ClassValue>('');
+  readonly zActionClick = output<void>();
 
+  protected readonly descriptionId = computed(() => (this.zDescription() ? generateId('card-description') : ''));
+  protected readonly titleId = computed(() => (this.zTitle() ? generateId('card-title') : ''));
```
[the remark this answers is not in the document]
Comment (mikij) [reply libs/zard/src/lib/shared/components/card/card.component.ts]: so you are telling me that when signal changes host value dependent on signal will not change?

[the remark this answers is not in the document]
Comment (mikij) [reply libs/zard/src/lib/shared/components/card/card.component.ts]: okay now I understand. but is not it better then to not use computed at all and initialize ids only once. Then no matter what is the content of description or title binding will be unchangeable. I think empty string as id is irrelevant here

File: libs/zard/src/lib/shared/components/card/demo/default.ts [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+import { generateId } from '@/shared/utils/merge-classes';
 
 @Component({
   selector: 'z-demo-card-default',
   imports: [ZardCardComponent, ZardButtonComponent],
-  standalone: true,
   template: `
     <z-card
-      class="w-full max-w-sm"
+      class="w-full md:w-94"
```
[the remark this answers is not in the document]
Comment (mikij) [reply libs/zard/src/lib/shared/components/card/demo/default.ts:12]: for Tailwind 4 that is unnecessary

[the remark this answers is not in the document]
Comment (mikij) [reply libs/zard/src/lib/shared/components/card/demo/default.ts:12]: you should start using 
{
  "mcpServers": {
    "tailwindcss-server": {
      "command": "npx",
      "args": ["-y", "tailwindcss-mcp-server"]
    }
  }
}
